src/tests/ftest/util/command_utils_base.py

```diff
@@ -393,27 +419,63 @@ def get_yaml_data(self):
 
         return yaml_data if self.title is None else {self.title: yaml_data}
 
+    def is_yaml_data_updated(self):
+        """Determine if any of the yaml file parameters have been updated.
+
+        Returns:
+            bool: whether or not a yaml file parameter has been updated
+
+        """
+        yaml_data_updated = False
+        if isinstance(self.other_params, YamlParameters):
+            yaml_data_updated = self.other_params.is_yaml_data_updated()
+        if not yaml_data_updated:
+            for name in self.get_param_names():
+                if getattr(self, name).updated:
+                    yaml_data_updated = True
+                    break
+        return yaml_data_updated
+
+    def reset_yaml_data_updated(self):
+        """Reset each yaml file parameter updated state to False."""
+        if isinstance(self.other_params, YamlParameters):
+            self.other_params.reset_yaml_data_updated()
+        for name in self.get_param_names():
+            getattr(self, name).updated = False
+
     def create_yaml(self, filename=None):
         """Create a yaml file from the parameter values.
 
+        A yaml file will only be created if at least one of its parameter values
+        have be updated (BasicParameter.updated = True).
+
         Args:
             filename (str, optional): the yaml file to generate with the
                 parameters. Defaults to None, which uses self.filename.
 
         Raises:
             CommandFailure: if there is an error creating the yaml file
 
-        """
-        if filename is None:
-            filename = self.filename
-        yaml_data = self.get_yaml_data()
-        self.log.info("Writing yaml configuration file %s", filename)
-        try:
-            with open(filename, 'w') as write_file:
-                yaml.dump(yaml_data, write_file, default_flow_style=False)
-        except Exception as error:
-            raise CommandFailure(
-                "Error writing the yaml file {}: {}".format(filename, error))
+        Returns:
+            bool: whether or not an updated yaml file was created
+
+        """
+        create_yaml = self.is_yaml_data_updated()
+        if create_yaml:
+            # Write a new yaml file if any of the parameters have been updated
+            if filename is None:
+                filename = self.filename
+            yaml_data = self.get_yaml_data()
+            self.log.info("Writing yaml configuration file %s", filename)
+            try:
+                with open(filename, 'w') as write_file:
+                    yaml.dump(yaml_data, write_file, default_flow_style=False)
+            except Exception as error:
+                raise CommandFailure(
```

Comment:
(pylint-raise-missing-from) Consider explicitly re-raising using the 'from' keyword